spec/lib/schedule_spec.rb

```diff
@@ -44,26 +45,21 @@
       # 24<25>26 27 28 29 30
       # 31
 
-      start_date = Date.new(2014, 8, 18) # Monday
-      expect(subject.dates(start_date).first).to eq(start_date + 4) # Friday
-
-      start_date = Date.new(2014, 8, 19) # Tuesday
-      expect(subject.dates(start_date).first).to eq(start_date + 4) # Saturday
-
-      start_date = Date.new(2014, 8, 20) # Wednesday
-      expect(subject.dates(start_date).first).to eq(start_date + 7) # Wednesday
-
-      start_date = Date.new(2014, 8, 21) # Thursday
-      expect(subject.dates(start_date).first).to eq(start_date + 7) # Thursday
-
-      start_date = Date.new(2014, 8, 22) # Friday
-      expect(subject.dates(start_date).first).to eq(start_date + 7) # Friday
-
-      start_date = Date.new(2014, 8, 23) # Saturday
-      expect(subject.dates(start_date).first).to eq(start_date + 6) # Friday
-
-      start_date = Date.new(2014, 8, 24) # Sunday
-      expect(subject.dates(start_date).first).to eq(start_date + 5) # Friday
+      week_start = monday = Date.new(2014, 8, 18)
+      first_bookable_visits = {
+        :monday => week_start.next_day(Date::DAYS_INTO_WEEK[:friday]),
+        :tuesday => week_start.next_day(Date::DAYS_INTO_WEEK[:saturday]),
+        :wednesday => week_start.next_week(:wednesday),
```

Comment:
Probably, but [these sorts of issues]( are exactly what led me to look at this code in the first place. It turns out that the code doesn't behave at all as we intend, so this change is purely to clarify what the code *actually does*, so when we come to make the change in future, there's a good example of how to communicate that change in intent.

I should put that in the commit message and PR description! (I will, when you're done.)